wgengine/router_freebsd.go

```diff
@@ -139,15 +145,7 @@ func (r *freebsdRouter) SetRoutes(rs RouteSettings) error {
 }
 
 func (r *freebsdRouter) Close() error {
-	out, err := cmd("ifconfig", r.tunname, "down").CombinedOutput()
-	if err != nil {
-		r.logf("running ifconfig failed: %v\n%s", err, out)
-	}
-
-	if err := r.restoreResolvConf(); err != nil {
-		r.logf("failed to restore system resolv.conf: %v", err)
-	}
-
+	r.device.Close()
```

Comment:
Hmm. On linux, the router is not in charge of closing the device. Is that because some caller is doing it? Or are we being bad and forgetting to close it on linux? I'm digging into the code now, but that's the only thing I worry about in this change.

Reply:
Okay, digging done. It looks like wireguard-go/device takes ownership of the tun device, and Close()ing the wgdev would also close the tundev.

The problem is that Close() in userspace.go doesn't close the wgdev.

So we definitely have confused ownership in here. I _think_ the correct fix is to `e.wgdev.Close()` in `userspaceEngine.Close()`, because userspaceEngine is owning and doing stuff with wgdev - so it should be the one doing orderly shutdown.

Can you move the Close() into userspace.go, so that all platforms clean up the tun device correctly? Then FreeBSD should just work also.

Reply:
Sure thing... looks like some tests are failing so I'll play around with the implementation a bit

Reply:
Looks like fakeTun just needed the evchan to be closed.  Might want to check it out on your end but it should be good to go.